### ISSUE TITLE
mpdf : Sign Created PDF using RSA/PKI Certificate

### ISSUE THREAD

**Issue by tetrahydra** [mpdf/mpdf#389]
Hello,

I would like to know if it is possible to sign generated PDF using RSA/PKI Certificate?

mPDF 7.0.0
PHP 7

**Comment by JordiRieraWirth** [comment]
Hello tetrahydra.

have you managed to sign with mpdf or another?

**Comment by StalkAlex** [comment]
@finwe what's the status of this feature? I see referenced issue above where you mentioned that it would be next beta release, then you closed that PR as "it doesn't make sense", then you're asking to create new issue to request new feature. 

So this issue still opened more then a year and I wanted to ask what's the current situation on this feature? Haven't it implemented yet or there is another recommended solution?

**Comment by finwe** [comment]
@StalkAlex See Labels. This is an open feature request waiting for someone to put time into it.

The referenced issue (PR, actually) does not make sense - its commits and comments reference some other topic and are just copied there because the PR was based on an incorrect base.

**Comment by StalkAlex** [comment]
@finwe  ok, thanks for clarification!

**Comment by subpardaemon** [comment]
hi,
if i'm reading this correctly this is still an open issue and a tough one -- and the PR referenced didn't contain any work on it? so it is still up for grabs?
how hard can it be? can it be deduced from tcpdf?

**Comment by finwe** [comment]
No work that I know of. Feel free to chip in.

I guess it could be deduced from tcpdf, but beware - in the recent attempt to [convert mPDF to a more permissive licence than GPL](https://github.com/mpdf/mpdf/issues/832) we cannot reuse any of TCPDF code.

**Comment by subpardaemon** [comment]
that's ok, i like to do my own coding. :) could you help as far as to quickly summarize the internals of mpdf, regarding where i could step in with the signing process?

i know that in broad strokes, i will only be able to do the actual signing when the actual pdf building takes place, any way there is a hook in the build process that i could use for that purpose?

it's a bit tricky as the placeholder for the clickable "signed" area needs to be put in place before the pdf is built from html, but the pdf can only be signed once at least the basic structure is ready (to calculate hashes, i'm guessing, to be able to claim the document has not been tampered with).

**Comment by finwe** [comment]
I'd guess somewhere at the start of the Output method, where the document content is already complete and won't change.

**Comment by subpardaemon** [comment]
also, how deep is the integration with FPDI? is that only used to bring in pdf documents, or are other parts of it being used in pdf generation?

**Comment by claymore008** [comment]
also no change to create signed PDFs?

**Comment by ikiK-CRO** [comment]
Bump ;)
